theorafile.c

```diff
@@ -350,14 +366,20 @@ void tf_close(OggTheora_File *file)
 	}
 	if (file->vpackets)
 	{
-		ogg_stream_clear(&file->vstream);
+		ogg_stream_clear(&file->vstream[file->vtrack]);
 	}
 
 	/* Metadata */
 	th_info_clear(&file->tinfo);
 	th_comment_clear(&file->tcomment);
-	vorbis_comment_clear(&file->vcomment);
-	vorbis_info_clear(&file->vinfo);
+	for (int i = 0; i < file->vtracks; i++)
```

Comment:
```suggestion
	for (i = 0; i < file->vtracks; i += 1)
```

i has to be declared at the top for VS2010.